src/gsumm.c

```diff
@@ -1268,6 +1268,7 @@ SEXP gshift(SEXP x, SEXP nArg, SEXP fillArg, SEXP typeArg) {
     copyMostAttrib(x, tmp); // needed for integer64 because without not the correct class of int64 is assigned
   }
   UNPROTECT(nprotect);
-  return(ans);
+  // consistency with plain shift(): "strip" the list in the 1-input case, for convenience
+  return isVectorAtomic(x) && length(ans) == 1 ? VECTOR_ELT(ans, 0) : ans;
```

Comment:
@ben-schwen do we actually need the `isVectorAtomic()` check in the `gshift` case? Is it possible to get a `list()` here in gshift?

Reply:
Right now no, the only supported types are the six atomic vector types. AFAIR I deactivated the support for list columns because we use `coerceAs` for the fill argument and `coerceAs` only supports atomic vectors.

Reply:
I'll leave the `isVectorAtomic()` check, simpler to keep consistency with "plain" shift.